Gems/Terrain/Assets/Shaders/Terrain/TerrainDetailHelpers.azsli

```diff
@@ -225,26 +250,105 @@ void GetDebugDetailSurface(inout DetailSurface surface, uint material1, uint mat
     surface.m_height = 0.5;
 }
 
-bool GetDetailSurface(inout DetailSurface surface, float2 idUv, float2 uv)
+void BlendDetailMaterial(
+    inout DetailSurface surface,
+    inout DetailSurface tempSurface,
+    uint material1, uint material2, float blend, float2 detailUv, float weight)
 {
-    uint4 material1 = TerrainSrg::m_detailMaterialIdImage.GatherRed(TerrainSrg::DetailSampler, idUv, 0).xyzw;
-    uint4 material2 = TerrainSrg::m_detailMaterialIdImage.GatherGreen(TerrainSrg::DetailSampler, idUv, 0).xyzw;
+    GetDetailSurfaceForMaterial(tempSurface, material1, detailUv);
+    if (material2 != 0xFF)
+    {
+        WeightDetailSurface(tempSurface, weight * blend);
+        AddDetailSurface(surface, tempSurface);
+
+        GetDetailSurfaceForMaterial(tempSurface, material2, detailUv);
+        WeightDetailSurface(tempSurface, weight * (1.0 - blend));
+        AddDetailSurface(surface, tempSurface);
+    }
+    else
+    {
+        WeightDetailSurface(tempSurface, weight);
+        AddDetailSurface(surface, tempSurface);
+    }
+}
 
+bool GetDetailSurface(inout DetailSurface surface, float2 detailMaterialIdUv, float2 detailUv)
```

Comment:
Suggestion:  It would be SUPER nice to have a comment block above this explaining the general algorithm in this method.  All the lines of code make sense, but it's hard to see at a glance what it's trying to accomplish.